### DISCUSSION TITLE
What about refactoring support in pom.xml

### DISCUSSION THREAD

**Discussion by ahoehma** [eclipse-m2e/m2e-core#1175] [Ideas]
Hi all,

would it be cool to have something like:

1. mark a hardcoded dependency and execute "extract version to version-property" (would be really cool in a multi-module project)
2. mark a dependency and execute "move version to dependency-management" (for multi-module may cool)
3. mark a dependency and execute "pull up" (move dep def. to parent pom)
4. or something like "extract dependency-management" (incl. version-properties) and may "as new parent-pom"
5. format pom.xml basend on a template (order of the different sections: parent, company stuff, properties, dep-mgt, deps, build, ..)

I'm just asking :)

**Comment by mickaelistria** [comment]
All those should be requested as individual enhancements odeas to the [lemminx-maven issue tracker](https://github.com/eclipse/lemminx-maven). LemMinX-Maven is the brick that provides the edition capabilities and having it improved also allows to improve support for pom edition in other IDEs that rely on it (such as VSCode)
